examples/benchmarks.rs

```diff
@@ -0,0 +1,382 @@
+            osc.start_at(offset);
+            osc.stop_at(offset + 1.); // why not 0.1 ?
+
+            offset += 140. / 60. / 4.; // 140 bpm (?)
+        }
+
+        benchmark(name, &mut context, &mut results);
+
+        write!(stdout, ".").unwrap();
+        stdout.flush().unwrap();
+    }
+
+    {
+        let name = "Sawtooth with automation";
+
+        let mut context =
+            OfflineAudioContext::new(1, DURATION * sample_rate.0 as usize, sample_rate);
+
+        let osc = context.create_oscillator();
+        osc.connect(&context.destination());
+        osc.set_type(OscillatorType::Sawtooth);
+        osc.frequency().set_value(2000.);
+        osc.frequency().linear_ramp_to_value_at_time(20., 10.);
+        osc.start_at(0.);
+
+        benchmark(name, &mut context, &mut results);
+
+        write!(stdout, ".").unwrap();
+        stdout.flush().unwrap();
+    }
+
+    // -------------------------------------------------------
+    // display results
+    // -------------------------------------------------------
+    let stdin = stdin();
+    let stdin = stdin.lock();
+
+    write!(stdout, "\r\n\r\n").unwrap();
+
+    write!(
+        stdout,
+        "{}+ index {}{}| name {}{}| duration (ms) {}{}| Speedup vs. realtime {}\r\n",
+        termion::style::Bold,
+        cursor::Left(200),
+        cursor::Right(10),
+        cursor::Left(200),
+        cursor::Right(65),
+        cursor::Left(200),
+        cursor::Right(65 + 16),
+        termion::style::Reset,
+    )
+    .unwrap();
+
+    for (index, result) in results.iter().enumerate() {
+        write!(
+            stdout,
+            "- {} {}{}| {} {}{}| {} {}{}| {} \r\n",
+            index + 1,
+            cursor::Left(200),
+            cursor::Right(10),
+            result.name,
+            cursor::Left(200),
+            cursor::Right(65),
+            result.duration.as_millis(),
+            cursor::Left(200),
+            cursor::Right(65 + 16),
+            (DURATION as u128 * 1000) / result.duration.as_millis(),
+        )
+        .unwrap();
+    }
+
+    write!(stdout, "\r\n").unwrap();
+    // @todo - this needs to be reviwed can only play 9 first buffers...
+    write!(
+        stdout,
+        "+ Press [1-9] to play output buffer for each test\r\n"
+    )
+    .unwrap();
+    write!(stdout, "+ Press \"s\" to stop playback\r\n").unwrap();
+    write!(stdout, "+ Press \"q\" to quit\r\n").unwrap();
+    write!(stdout, "\r\n").unwrap();
+
+    stdout.flush().unwrap();
+
+    // -------------------------------------------------------
+    // handle input and preview
+    // -------------------------------------------------------
+    let context = AudioContext::new(None);
+    let mut current_source: Option<AudioBufferSourceNode> = None;
+
+    // that's really dirty
+    let mut bytes = stdin.bytes();
+    loop {
+        let b = bytes.next().unwrap().unwrap();
+
+        match b {
+            // quit
+            b'q' => return,
```

Comment:
Could you also listen for ctrl-c here? Was a bit surprised I could not kill the program.
Other than that, lets merge!

Reply:
it's there!
I have also improved the interface and interaction, let me know what you think